### PR TITLE
Feature/improve travis

### PR DESCRIPTION
**Pull request by mauriciovieira**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+all: test
+
+install-ci:
+	pip install -r requirements/ci.txt
+
+install-local:
+	pip install -r requirements/local.txt
```
Comment (mauriciovieira) [inline Makefile]: Wrong. This makefile does not babysit the user. I am using virtualenv here.

Comment (mauriciovieira) [inline Makefile]: I am using [virtualenvwrapper](https://virtualenvwrapper.readthedocs.io/en/latest/), so my virtualenv is not on venv/

You can add a make task to activate virtualenv, anyways, if you want.

Comment (mauriciovieira) [inline Makefile]: But bear in mind that `pip install` does not have to know if it's in a virtualenv or not. For the sake of Continuous Integration, this makefile is enough. See the travis outputs on ✅

File: setup.sh [hunk 2]
```diff
@@ -10,6 +10,10 @@ echo "Creating virtual environment..."
 virtualenv -p $python venv
 source venv/bin/activate
 echo "Installing dependencies..."
-pip install -r requirements.txt
+pip install -r requirements/base.txt
```
Comment (mauriciovieira) [inline setup.sh]: This is not needed. Read `requirements.txt` file.

File: setup.sh [hunk 3]
```diff
@@ -10,6 +10,10 @@ echo "Creating virtual environment..."
 virtualenv -p $python venv
 source venv/bin/activate
 echo "Installing dependencies..."
-pip install -r requirements.txt
```
Comment (mauriciovieira) [inline setup.sh]: This is not needed, because `requirements.txt` requires `requirements/base.txt`

Comment (mauriciovieira) [inline setup.sh]: Also, I would not write such a big commit message. Read [5 Useful Tips For A Better Commit Message](https://robots.thoughtbot.com/5-useful-tips-for-a-better-commit-message)

File: Dockerfile [hunk 4]
```diff
@@ -3,8 +3,8 @@ FROM python:3-stretch
 RUN pip install uwsgi
 
 WORKDIR /app
-COPY requirements.txt /app/
-RUN pip install -r requirements.txt
```
Comment (mauriciovieira) [inline Dockerfile]: This is not needed, because requirements.txt requires requirements/base.txt

File: README.md [hunk 5]
```diff
@@ -19,7 +19,7 @@ A backend for storing OpenAPI specifications
 1. Install virtualenv
 2. Create a virtualenv (`virtualenv -p /path/to/python3 venv`)
 3. Enter the virtualenv (`source venv/bin/activate`)
-4. Install dependencies (`pip install -r requirements.txt`)
+4. Install dependencies (`pip install -r requirements/base.txt`)
```
Comment (mauriciovieira) [inline README.md]: This is not needed, because requirements.txt requires requirements/base.txt

File: setup.sh (hunk 3 above)
Comment (xylix) [inline setup.sh]: Thanks for the tip and sorry, i somehow missed the changes to requirements.txt. Should have been more careful when reading the changes.

File: README.md (hunk 5 above)
Comment (xylix) [inline README.md]: I'll undo these useless changes, sorry.

Comment (mauriciovieira) [inline README.md]: Alright 👍 . I'd just delete the commit and `push --force` to the branch.

File: Dockerfile (hunk 4 above)
Comment (mauriciovieira) [inline Dockerfile]: Well, you might need to copy requirements.txt *and* requirements dir with the other files to `/app/` here.

File: README.md (hunk 5 above)
Comment (mauriciovieira) [inline README.md]: No problem. Check the copying of all requirements files in docker.

File: Dockerfile (hunk 4 above)
Comment (xylix) [inline Dockerfile]: Yes you are correct. I will do that after undoing the  other changes.

**Comment by xylix** [comment]
I'm changing the Dockerfile and testing to build it right now, will take some time. Are you still planning to change something here or can I merge after changing the Dockerfile @mauriciovieira ?

**Comment by mauriciovieira** [comment]
@xylix One thing that can be done is to build the image after commit. Somewhat [like](https://github.com/apinf/platform/blob/794a2431d45b4b1e10dc845e050bfba572b65d09/.travis.yml#L49) [Platform](https://github.com/apinf/platform/blob/4a6d5f8e6cb8a07f39de99f29c598ddbab2c2063/.scripts/docker_build.sh).

Also, move the `*.sh` to a `scripts/` directory and add them to `Makefile` tasks. It's up to you.

**Comment by xylix** [comment]
I'll create another PR for the automatic docker image generation. Merging this now.

**Pull request merged**